### ISSUE TITLE
Encryption failed: unusable public key (gmime version?)

### ISSUE THREAD

**Issue by igoralmeida** [astroidmail/astroid#637]
Hi,
I am getting this "encryption failed" error below. The only thing I can think of is that I recently recompiled notmuch as well, which seems to now use gmime3 (which I blindly installed using pacman).

I confirmed that I have `crypto.gpg.always_trust` set to `true`, and that this works:
```
echo hi | gpg -e -r 'Gaute Hope <eg@gaute.vetsj.com>' -r 'Igor Almeida <igor.contato@gmail.com>' -a --trust-model=always | gpg -d
```

Any thoughts?

```
[22:24:02] [0x00007f5da6e3b400] [E] [debug] set info: Edit message with 'Enter'.
[22:24:02] [0x00007f5da6e3b400] [E] [debug] ae: sending: Ack
[22:24:02] [0x00007f5da6e3b400] [E] [debug] ae: send (async) waiting for lock
[22:24:02] [0x00007f5da6e3b400] [E] [debug] ae: wrote: 51 of 51 bytes.[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ae: send (sync) ACK received.

[22:24:02] [0x00007f6b42ec4a00] [M] [debug] pc: got ack (s: true) , focus: 1559006629.kc1z87muro.astroid@arch-blu.none, e: 0
[22:24:02] [0x00007f5da6e3b400] [E] [debug] ae: send (async) message sent.
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] cm: initialize..
[22:24:02] [0x00007f6b42ec4a00] [M] [info] msg: loading message from file (mid supplied): /home/igor/.cache/astroid/1559006629.kc1z87muro.astroid@arch-blu.none
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] chunk (173): content-type: text/plain
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] chunk: is part (viewable: true, attachment: false) 
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] chunk: body: part
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] chunk: plain text (out html: false)
[22:24:02] [0x00007f6b42ec4a00] [M] [warning] charset: not defined.
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ms: deconstruct
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] chunk: deconstruct.
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] same account: true
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] cm: build..
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] cm: adding inline signature from: /home/igor/mail/gmail-signature.txt
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] cm: finalize..
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] crypto: gpg: gpg2
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] cr: encrypting for: 
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] eg@gaute.vetsj.com 
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] igor.contato@gmail.com 
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] crypto: failed to encrypt message: Encryption failed: Unusable public key

(astroid:29434): gmime-CRITICAL **: 22:24:02.816: g_mime_message_set_mime_part: assertion 'GMIME_IS_OBJECT (mime_part)' failed

(astroid:29434): GLib-GObject-CRITICAL **: 22:24:02.816: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
[22:24:02] [0x00007f6b42ec4a00] [M] [error] cm: failed encrypting or signing: Encryption failed: Unusable public key
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] crypto: deconstruct.
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] em: set warning (ready: true): Failed encrypting: Encryption failed: Unusable public key
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] tv: set warning: Failed encrypting: Encryption failed: Unusable public key
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ae: sending: Info
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ae: send (sync) waiting for lock..
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ae: send (sync) lock acquired.
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ae: wrote: 108 of 108 bytes.
[22:24:02] [0x00007f6b42ec4a00] [M] [debug] ae: send (sync) waiting for ACK..
[22:24:02] [0x00007f5da6e3b400] [E] [debug] set warning: Failed encrypting: Encryption failed: Unusable public key
[22:24:02] [0x00007f5d56765700] [E] [debug] reader waiting..
[22:24:02] [0x00007f5da6e3b400] [E] [debug] ae: sending: Ack
[22:24:02] [0x00007f5da6e3b400] [E] [debug] ae: send (async) waiting for lock
```

**Comment by gauteh** [comment]
Seems like perhaps gpg via astroid might try to use an expired key?

tir. 28. mai 2019 kl. 03:50 skrev Igor Almeida <notifications@github.com>:

**Comment by gauteh** [comment]
Is this caused by missing `always_trust`?

**Comment by gauteh** [comment]
Closed as duplicate of #638
